agent/src/main/java/io/pyroscope/javaagent/Profiler.java

```diff
@@ -169,6 +158,7 @@ final synchronized Snapshot dump() {
         if (format == Format.JFR) {
             restartJFR();
         } else {
+            instance.stop();
```

Comment:
in Format.JFR, dumpJFR() method calls `instance.stop()`. But  Format.COLLAPSED don't call it.
https://github.com/pyroscope-io/pyroscope-java/blob/main/agent/src/main/java/io/pyroscope/javaagent/Profiler.java#L198

`AsyncProfiler#start` throws `IllegalStateException If profiler is already running` if it's already started.
https://github.com/jvm-profiling-tools/async-profiler/blob/91eab916343fe646f6ad266694312c9175f28e0f/src/api/one/profiler/AsyncProfiler.java#L57

In async-profiler 2.0, it's the spec. But async-profiler never throws exception at async-profiler 2.0.
https://github.com/jvm-profiling-tools/async-profiler/blob/v2.0/src/api/one/profiler/AsyncProfiler.java#L57